src/ru/animelib/src/eu/kanade/tachiyomi/animeextension/ru/animelib/Animelib.kt

```diff
@@ -0,0 +1,408 @@
+            }
+        }.also(screen::addPreference)
+    }
+
+    // =============================== Details ===============================
+    override fun animeDetailsRequest(anime: SAnime): Request {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("anime")
+        url.addPathSegment(anime.url)
+        url.addQueryParameter("fields[]", "genres")
+        url.addQueryParameter("fields[]", "summary")
+        url.addQueryParameter("fields[]", "authors")
+        url.addQueryParameter("fields[]", "publisher")
+        url.addQueryParameter("fields[]", "otherNames")
+        url.addQueryParameter("fields[]", "anime_status_id")
+
+        return GET(url.build())
+    }
+
+    override fun getAnimeUrl(anime: SAnime) = "$baseUrl/${anime.url}"
+
+    override fun animeDetailsParse(response: Response): SAnime {
+        val animeInfo = json.decodeFromString<AnimeInfo>(response.body.string())
+
+        return animeInfo.data.toSAnime()
+    }
+
+    // =============================== Episodes ===============================
+    override fun episodeListRequest(anime: SAnime): Request {
+        val url = apiUrl.toHttpUrl().newBuilder()
+        url.addPathSegment("episodes")
+        url.addQueryParameter("anime_id", anime.url)
+
+        return GET(url.build())
+    }
+
+    override fun episodeListParse(response: Response): List<SEpisode> {
+        val episodeList = json.decodeFromString<EpisodeList>(response.body.string())
+
+        return episodeList.data.map { it.toSEpisode() }.reversed()
+    }
+
+    // =============================== Video List ===============================
+    override fun videoListParse(response: Response): List<Video> {
+        val episodeData = json.decodeFromString<EpisodeVideoData>(response.body.string())
+        val videoServer = fetchPreferredVideoServer()
+
+        val videos = mutableListOf<Video>()
+        if (episodeData.data.players.isNullOrEmpty()) {
+            return videos
+        }
+
+        for (videoInfo in episodeData.data.players) {
+            val playerName = videoInfo.player.lowercase()
+            if (playerName == "kodik") {
+                // TODO maybe in future
+                // videos.addAll(kodikVideoLink(videoInfo.src))
+            } else if (playerName == "animelib") {
+                videos.addAll(animelibVideoLinks(videoInfo, videoServer))
+            }
+        }
+
+        // Filter by dub team preference
+        val prefDubTeam = preferences.getString(PrefConstants.PREF_DUB_TEAM_KEY, "")
+        val teams = prefDubTeam?.split(',')
+
+        if (teams.isNullOrEmpty()) {
+            return videos
+        }
+
+        videos.removeAll {
+            var remove = true
+            for (team in teams) {
+                if (it.quality.contains(team, true)) {
+                    remove = false
+                    break
+                }
+            }
+
+            remove
+        }
+
+        return videos
```

Comment:
```suggestion
        val videoList = episodeData.data.players?.flatMap { videoInfo ->
            val playerName = videoInfo.player.lowercase()
            when (playerName) {
                "kodik" -> emptyList() // TODO maybe in future
                "animelib" -> animelibVideoLinks(videoInfo, videoServer)
                else -> emptyList()
            }
        } ?: emptyList()

        val teams = preferences.getString(PrefConstants.PREF_DUB_TEAM_KEY, "")?.split(',')
            ?: return videoList
        return videoList.filter { video ->
            teams.any { video.quality.contains(it, true) }
        }
```
Maybe this is more of a nitpick, but could be nice to not use mutable lists

Reply:
Thanks for idea. I slightly changed it by filtering before parsing video urls. Kodik player needs some additional requests for each video, so its better to skip this additional requests if user doesn't need kodik video